### PR TITLE
Kotlin conversion: DateUtil

### PR DESCRIPTION
**Pull request by sharvaniharan**

### CODE REVIEW

File: app/src/main/java/org/wikipedia/util/DateUtil.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    // TODO: Switch to DateTimeFormatter when minSdk = 26.
+    @JvmStatic
+    @Synchronized
+    fun iso8601DateFormat(date: Date?): String {
+        return getCachedDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ROOT, true)!!.format(date!!)
+    }
+
+    @JvmStatic
+    @Synchronized
+    @Throws(ParseException::class)
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/util/DateUtil.kt:25]: I don't think we need to explicitly specify `@Throws` annotations anymore.

File: app/src/main/java/org/wikipedia/util/DateUtil.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import android.text.format.DateFormat
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.feed.model.UtcDate
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.*
+
+object DateUtil {
+    private val DATE_FORMATS: MutableMap<String, SimpleDateFormat> = HashMap()
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/util/DateUtil.kt:14]: Can be simplified to:
`private val DATE_FORMATS = HashMap<String, SimpleDateFormat>()`

**Review by dbrant — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/util/DateUtil.kt [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    @JvmStatic
+    @Synchronized
+    fun iso8601DateFormat(date: Date): String {
+        return getCachedDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ROOT, true)!!.format(date)
+    }
+
+    @JvmStatic
+    @Synchronized
+    fun iso8601DateParse(date: String): Date {
+        return getCachedDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ROOT, true)!!.parse(date!!)!!
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/util/DateUtil.kt:26]: Please remove `!!` on `date`.

**Review by cooltey — COMMENTED** [review]
One minor comment

**Pull request merged**